### PR TITLE
Test/249 convert js files to ts

### LINKED ISSUE #249: Convert Existing Cypress tests to Typescript
Currently all the Cypress step definitions and page classes are written in Javascript. 

It would be good to convert all of them to Typescript. 

- Possible issues with custom Cypress commands and conflicts with Jest.

https://docs.cypress.io/guides/tooling/typescript-support#Install-TypeScript

### PR DESCRIPTION
**Pull request by jtsang586**
Convert existing Cypress page classes and step definitions from Javascript to Typescript

Also updated Cypress to latest version (9.5.4)

Closes #249

### CODE REVIEW

File: libs/cypress/src/lib/commands/get-by-test-id.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
-// eslint-disable-next-line @typescript-eslint/no-namespace
```
Comment (dexturr) [inline libs/cypress/src/lib/commands/get-by-test-id.ts:1]: Question (non-blocking): Why were these changes needed?

File: package.json [hunk 2]
```diff
@@ -102,7 +102,7 @@
     "@typescript-eslint/parser": "~5.10.0",
     "babel-jest": "27.2.3",
     "babel-loader": "8.1.0",
-    "cypress": "^9.5.1",
+    "cypress": "9.5.4",
```
Comment (dexturr) [inline package.json:105]: Question (blocking) why moving to a fixed version?

**Review by dexturr — COMMENTED** [review]
Lgtm one blocking question

**Review by sam-keen — APPROVED** [review]
Apart from Dex's points, looks great

**Comment by dexturr** [comment]
Looks like after stats merge this needs a rebase. Let's get this in as the next PR as priority, as we don't want to end up doing this multiple times

File: apps/trading-e2e/src/support/step_definitions/deal-ticket.step.ts [hunk 3]
```diff
@@ -4,27 +4,27 @@ import DealTicket from '../trading-windows/deal-ticket';
 const dealTicket = new DealTicket();
 
 When('I place a buy {string} market order', (orderType) => {
-  dealTicket.placeMarketOrder(true, 100, orderType);
+  dealTicket.placeMarketOrder(true, '100', orderType);
```
[the remark this answers is not in the document]
Comment (jtsang586) [reply apps/trading-e2e/src/support/step_definitions/deal-ticket.step.ts:7]: I can't take any credit for this. Cy.type only accepts string as parameter :stuck_out_tongue:

File: libs/cypress/src/lib/commands/get-by-test-id.ts (hunk 1 above)
Comment (jtsang586) [reply libs/cypress/src/lib/commands/get-by-test-id.ts:1]: So for some reason the Cypress custom commands don't get recognised when the files are in typescript unless the commands are declared globally. 
https://github.com/cypress-io/cypress-documentation/issues/2565#issuecomment-651251129

**Review by dexturr — APPROVED** [review]
